dog.md

```diff
@@ -0,0 +1,208 @@
+```
+
+Dog Events
+----------
+
+```k
+    syntax CustomEvent ::= Bark(ilk: String, urn: Address, ink: Wad, art: Wad, due: FInt, clip: Address, Id: Int) [klabel(Bark), symbol]
+ //-----------------------------------------------------------------------------------------------------------------------------------
+
+    syntax DogStep ::= "emitBark" String Address Wad Wad FInt Address Int
+ // --------------------------------------------------------
+    rule <k> BARK_ID:Int ~> emitBark ILK URN INK ART DUE CLIP _ => BARK_ID ... </k>
+         <return-value> BARK_ID:Int </return-value>
+         <frame-events> ... (.List => ListItem(Bark(ILK, URN, INK, ART, DUE, CLIP, BARK_ID))) </frame-events>
+```
+
+File-able Fields
+----------------
+
+These parameters are controlled by governance:
+
+-   `vow`:  Debt Engine
+-   `hole`: Max DAI needed to cover debt+fees of active auctions
+-   `chop`: Liquidation Penalty
+-   `clip`: Liquidator
+
+```k
+    syntax DogAuthStep ::= "file" DogFile
+ // -------------------------------------
+
+    syntax DogFile ::= "vow"  Address
+                     | "Hole" Rad
+                     | "hole" String Rad
+                     | "chop" String Wad
+                     | "clip" String Address
+ // -----------------------------------------
+    rule <k> Dog . file vow DOG_VOW => . ... </k>
+         <dog-vow> _ => DOG_VOW </dog-vow>
+
+    rule <k> Dog . file Hole HOLE => . ... </k>
+         <dog-hole> _ => HOLE </dog-hole>
+
+    rule <k> Dog . file hole ILK HOLE => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+      requires CHOP >=Wad wad(1)
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+         <clip-ilk> CLIP_ILK </clip-ilk>
+      requires ILK ==K CLIP_ILK
+```
+
+Dog Semantics
+-------------
+
+```k
+    syntax DogStep ::= "bark" String Address Address
+ // -----------------------------------------------
+    rule <k> Dog . bark ILK URN KPR
+    =>  #let ROOM = minRad( ( DOG_HOLE -Rad DOG_DIRT), (HOLE -Rad  DIRT) ) #in (
+        #let DART_INITIAL = minWad( ART, ( ( ROOM /Rate RATE) /Wad CHOP) ) #in (
+        #let DART_FINAL = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then ART #else   DART_INITIAL #fi) #in (
+        #let NON_DUSTY = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then true #else (DART_INITIAL *Rate RATE) >=Rad DUST  #fi) #in (
+        #let DINK = (INK *Wad DART_FINAL) /Wad ART #in (
+        #let DUE = DART_FINAL *Rate RATE #in (
+        #let TAB = DUE *RadWad2Rad CHOP #in (
+    NON_DUSTY
+    ~> KPR
+    ~> TAB
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0 ) ) ) ) ) ) )
```

Comment:
Shouldn't the last argument of `emitBark` here be the auction id, not a hardcoded `0`?